src/zampy/datasets/land_cover.py

```diff
@@ -134,16 +135,30 @@ def load(
                 )
                 raise ValueError(msg)
         files = list((ingest_dir / self.name).glob(f"{self.name}_*.nc"))
-
         ds = xr.open_mfdataset(files, chunks={"latitude": 200, "longitude": 200})
         ds = ds.sel(time=slice(time_bounds.start, time_bounds.end))
 
         grid = xarray_regrid.create_regridding_dataset(
             utils.make_grid(spatial_bounds, resolution)
         )
-        ds = ds.regrid.most_common(grid, time_dim="time", max_mem=1e9)
 
-        return ds
+        ds_regrid = {}
+        for variable in variable_names:
+            # select the variable to be regridded
+            da = ds[variable]
+
+            # get values for most common method
+            if "flag_values" in da.attrs:
+                regrid_values = da.attrs["flag_values"]
+            else:
+                regrid_values = np.unique(da.values)
+
+            da_regrid = da.regrid.most_common(grid, values=regrid_values)
+
+            # make sure dtype is the same
+            ds_regrid[variable] = da_regrid.astype(da.dtype)
```

Comment:
Will not be needed anymore in the next xarray-regrid release (but I'm not sure when exactly that'll be)

Reply:
I can add a comment about it.